editor/editor_properties_array_dict.cpp

```diff
@@ -566,6 +587,70 @@ void EditorPropertyArray::setup(Variant::Type p_array_type, const String &p_hint
 	}
 }
 
+void EditorPropertyArray::_reorder_button_gui_input(const Ref<InputEvent> &p_event) {
+	if (reorder_move_from_idx < 0) {
+		return;
+	}
+
+	Ref<InputEventMouseMotion> mm = p_event;
+	if (mm.is_valid()) {
+		Variant array = object->get_array();
+		int size = array.call("size");
+
+		if ((reorder_move_to_idx == 0 && mm->get_relative().y < 0.0f) || (reorder_move_to_idx == size - 1 && mm->get_relative().y > 0.0f)) {
+			return;
+		}
+
+		reorder_accumulated_y_motion += mm->get_relative().y;
+		float required_y_distance = 20.0f * EDSCALE;
+		if (ABS(reorder_accumulated_y_motion) > required_y_distance) {
+			int direction = reorder_accumulated_y_motion > 0.0f ? 1 : -1;
+			reorder_accumulated_y_motion -= required_y_distance * direction;
+
+			reorder_move_to_idx += direction;
+			if ((direction < 0 && reorder_move_to_idx % page_len == page_len - 1) || (direction > 0 && reorder_move_to_idx % page_len == 0)) {
+				page_slider->set_value(page_idx + direction); // automatically move to the next page
+			}
+			vbox->move_child(reorder_selected_property_hbox, reorder_move_to_idx % page_len + 2);
+			EditorNode::get_singleton()->get_inspector()->ensure_control_visible(reorder_selected_property_hbox);
+		}
+	}
+}
+
+void EditorPropertyArray::_reorder_button_down(int p_index) {
+	reorder_move_from_idx = p_index;
+	reorder_move_to_idx = p_index;
+	reorder_selected_property_hbox = Object::cast_to<HBoxContainer>(vbox->get_child(p_index % page_len + 2));
+	reorder_selected_button = Object::cast_to<Button>(reorder_selected_property_hbox->get_child(0));
```

Comment:
My original plan was to use [mouse_exited](https://docs.godotengine.org/en/stable/classes/class_control.html#signals) on the property hbox, but it was called at the wrong time (https://github.com/godotengine/godot/issues/16854) or not at all.My original plan was to use [mouse_exited](https://docs.godotengine.org/en/stable/classes/class_control.html#signals) on the property hbox, but it was called at the wrong time (https://github.com/godotengine/godot/issues/16854) or not at all.